Amplify/Categories/DataStore/Model/JSONHelper/JSONValueHolder.swift

```diff
@@ -27,11 +27,20 @@ public protocol JSONValueHolder {
     ///
     /// If a particular key has nil as it value, this method should return .some(nil) as the value.
     func jsonValue(for key: String) -> Any??
+
+    /// Return the value for the given key.
+    ///
+    /// If a particular key has nil as it value, this method should return .some(nil) as the value.
+    func jsonValue(for key: String, modelSchema: ModelSchema) -> Any??
 }
 
 extension JSONValueHolder {
```

Comment:
I'm having second thoughts about this extension. Since JSONValueHolder has a very particular use case (supporting Flutter data store), I'm inclined to make the conforming type implement both methods rather than provide conveniences which have unexpectedly `nil` behavior. Thoughts?

Reply:
Agreed and updated PR.